pkg/channel/consolidated/dispatcher/dispatcher.go

```diff
@@ -95,36 +95,8 @@ func NewDispatcher(ctx context.Context, args *KafkaDispatcherArgs) (*KafkaDispat
 	conf.Consumer.Return.Errors = true    // Returns the errors in ConsumerGroup#Errors() https://godoc.org/github.com/Shopify/sarama#ConsumerGroup
 	conf.Producer.Return.Successes = true // Must be enabled for sync producer
 
-	// Get the auth info
-	if args.KafkaAuthConfig != nil {
-		// tls
-		if args.KafkaAuthConfig.TLS != nil {
-			conf.Net.TLS.Enable = true
-			tlsConfig, err := source.NewTLSConfig(args.KafkaAuthConfig.TLS.Usercert, args.KafkaAuthConfig.TLS.Userkey, args.KafkaAuthConfig.TLS.Cacert)
-			if err != nil {
-				return nil, err
-			}
-			conf.Net.TLS.Config = tlsConfig
-		}
-		// SASL
-		if args.KafkaAuthConfig.SASL != nil {
-			conf.Net.SASL.Enable = true
-			conf.Net.SASL.Handshake = true
-
-			// if SASLTypeSCRAMSHA256 is provided we use that. Defaulting to SASLTypeSCRAMSHA512
-			if args.KafkaAuthConfig.SASL.SaslType == utils.SASLTypeSCRAMSHA256 {
-				conf.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient { return &source.XDGSCRAMClient{HashGeneratorFcn: source.SHA256} }
-				conf.Net.SASL.Mechanism = sarama.SASLTypeSCRAMSHA256
-
-			} else {
-				conf.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient { return &source.XDGSCRAMClient{HashGeneratorFcn: source.SHA512} }
-				conf.Net.SASL.Mechanism = sarama.SASLTypeSCRAMSHA512
-			}
-
-			conf.Net.SASL.User = args.KafkaAuthConfig.SASL.User
-			conf.Net.SASL.Password = args.KafkaAuthConfig.SASL.Password
-		}
-	}
+	// append the auth info:
+	source.UpdateSaramaConfigWithKafkaAuthConfig(conf, args.KafkaAuthConfig)
```

Comment:
if we return an error from  `UpdateSaramaConfigWithKafkaAuthConfig` below -- then we can catch it here and still `return nil, err` from `NewDispatcher` as well